receiver/prometheusexec/subprocessmanager/manager_test.go

```diff
@@ -0,0 +1,218 @@
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexec/subprocessmanager/config"
+)
+
+var (
+	getDelayTests = []struct {
+		name       string
+		elapsed    time.Duration
+		crashCount int
+		want       time.Duration
+	}{
+		{
+			name:       "healthy process",
+			elapsed:    15 * time.Second,
+			crashCount: 2,
+			want:       1 * time.Second,
+		},
+		{
+			name:       "healthy process",
+			elapsed:    15 * time.Hour,
+			crashCount: 6,
+			want:       1 * time.Second,
+		},
+		{
+			name:       "unhealthy process 1",
+			elapsed:    15 * time.Second,
+			crashCount: 4,
+		},
+		{
+			name:       "unhealthy process 2",
+			elapsed:    15 * time.Second,
+			crashCount: 5,
+		},
+		{
+			name:       "unhealthy process 3",
+			elapsed:    15 * time.Second,
+			crashCount: 6,
+		},
+		{
+			name:       "unhealthy process 4",
+			elapsed:    15 * time.Second,
+			crashCount: 7,
+		},
+	}
+
+	formatEnvSliceTests = []struct {
+		name     string
+		envSlice *[]config.EnvConfig
+		want     []string
+		wantNil  bool
+	}{
+		{
+			name:     "empty slice",
+			envSlice: &[]config.EnvConfig{},
+			want:     nil,
+			wantNil:  true,
+		},
+		{
+			name: "one entry",
+			envSlice: &[]config.EnvConfig{
+				{
+					Name:  "DATA_SOURCE",
+					Value: "password:username",
+				},
+			},
+			want: []string{
+				"DATA_SOURCE=password:username",
+			},
+			wantNil: false,
+		},
+		{
+			name: "three entries",
+			envSlice: &[]config.EnvConfig{
+				{
+					Name:  "DATA_SOURCE",
+					Value: "password:username",
+				},
+				{
+					Name:  "",
+					Value: "",
+				},
+				{
+					Name:  "john",
+					Value: "doe",
+				},
+			},
+			want: []string{
+				"DATA_SOURCE=password:username",
+				"=",
+				"john=doe",
+			},
+			wantNil: false,
+		},
+	}
+
+	runTests = []struct {
+		name        string
+		process     *Process
+		wantElapsed time.Duration
+		wantErr     bool
+	}{
+		{
+			name: "normal process 1, error process exit",
+			process: &Process{
+				Command: "go run ../testdata/test_crasher.go",
+				Port:    0,
+				Env: []config.EnvConfig{
+					{
+						Name:  "DATA_SOURCE",
+						Value: "username:password@(url:port)/dbname",
+					},
+				},
+				CustomName: "main",
+			},
+			wantElapsed: 4 * time.Millisecond,
+			wantErr:     false,
+		},
+		{
+			name: "normal process 2, normal process exit",
+			process: &Process{
+				Command: "go version",
+				Port:    0,
+				Env: []config.EnvConfig{
+					{
+						Name:  "DATA_SOURCE",
+						Value: "username:password@(url:port)/dbname",
+					},
+				},
+				CustomName: "main",
+			},
+			wantElapsed: 0 * time.Nanosecond,
+			wantErr:     false,
+		},
+		{
+			name: "shellquote error",
+			process: &Process{
+				Command:    "command flag='something",
+				Port:       0,
+				Env:        []config.EnvConfig{},
+				CustomName: "main",
+			},
+			wantElapsed: 0,
+			wantErr:     true,
+		},
+	}
+)
+
+func TestGetDelay(t *testing.T) {
```

Comment:
I'm a fan of keeping test cases inside the function that uses them. I definitely think related code should be close together